open-craft/learnupon-exporter
# [SE-5301] Scripts to export from edX to LearnUpon

## Description

Adds two new management commands for edX operators to export data for LearnUpon ingestion.

- manage.py lms learnupon_export_users output_directory course_id
- manage.py lms learnupon_export_enrollment_data output_directory course_id

Eg.

```bash
python manage.py lms learnupon_export_enrollment_data /edx/src/export "course-v1:edX+DemoX+Demo_Course"
```

Both commands create CSV files which can be (optionally) uploaded
to S3.

The following settings can be changed:

- `LEARNUPON_EXPORTER_STATIC_FILES_BUCKET = ''`
- `LEARNUPON_EXPORTER_STATIC_FILES_PATH = 'learnupon_exports/'`
- `LEARNUPON_EXPORTER_AWS_ACCESS_KEY_ID = ''`
- `LEARNUPON_EXPORTER_AWS_ACCESS_KEY_SECRET = ''`

## Supporting information

https://tasks.opencraft.com/browse/SE-5300

## Testing instructions

- Set up devstack
- Install the code with the comment `pip install git+https://github.com/open-craft/learnupon-exporter.git@keith/se_5301_create_export_scripts`
- You should be able to run the above management commands
- If the AWS fields are filled in the files will be uploaded to the value provided by `LEARNUPON_EXPORTER_STATIC_FILES_PATH`.

## Deadline

22 Feb 2022
